Feature/overview imp

sxe opened the pull request:
Hi, 
this merge request contains various changes. I tried to follow your coding style as close as possible but please be blunt when something is not up to your standards. I am happy to take critique and learn.

Changes:
 * Overview columns will jump to the next row on small screens and/or when the activity log is active.
 * Furnace and mining machines are now animated, to make it more obvious when they are active.
 * The normal rocket is now shown in the mining tile. (Just the normal rocket, since I don't have access to the other one yet)  
 * Oil amount in mining tile is now shown as an absolute value like "+2" to make it easier to read.
 * Geodes are now shown in the mining tile
 * Moonstones are now shown in the mining tile
 * Logs in the woodcutting tile are now clickable to add heat to the oven.
 * There is now a heat indicator in the woodcutting tile.
 * The tiles background color in the overview is now based on the "Left Menu Background Color" a user selects in the main idle-pixel settings.

I am not familiar with your versioning practices, so I just increased the "patch level" by one. Please let me know what you prefer. 

Let me know what you think. 
Cheers

sxe committed:
overview columns will jump to the next row on small screens.

sxe committed:
Add sublime config to git ignore

sxe committed:
add animated overview machines and furnace

sxe committed:
add rocket to overview. Not interactive yet.

sxe committed:
Refactor furnace image state change

sxe committed:
add rocket click event

sxe committed:
simplify oil amount display

sxe committed:
add geodes to mining overview tile

sxe committed:
make logs in woodcutting overview tile clickable, to add heat to the oven

sxe committed:
add heat indicator to woodoverview tile

sxe committed:
remove mining machine whitspace

sxe committed:
fix rocket at destination visuals

sxe committed:
add dynamic tile coloring to respect user color settings

sxe committed:
add moonstone to mining overview tile

sxe committed:
fix wrong overview background color and newly created accounts

kape142 commented on src/overview/mining/GeodeDisplay.tsx:
```diff
@@ -0,0 +1,42 @@
+  const [amount, setAmount] = useNumberItemObserver(geode + "_geode", "GeodeDisplay");
+
+  const onGeodeClick = (event: MouseEvent) => {
+    Modals.open_input_dialogue_with_value(geode + "_geode", "Open", "How many geodes to you want to crack?", amount, "CRACK_GEODE");
```
In general I want to avoid using modals when possible, and instead show how many will be consumed in a tooltip and let it be changed with shift and ctrl, like I have done with e.g. gathering bags.

If I were to implement this myself I would probably open all on a regular click, open half on shift-click, and open a small amount (like 5) on ctrl-click (like with gathering bags, bones etc.). I do think there is some value in having the option to open the modal, so if you think it should be kept, I'm open to e.g. regular click opening the modal, shift-click to open all geodes, and ctrl-click opening a small amount, or something like that.

sxe replied:
Makes total sense, in other places the modal is essential I think but for some reason I had it in my mind it was necessary here as well.  For geodes two modes make the most sense tome. So i implemented "All" and "All -1".

kape142 commented on src/overview/mining/GeodeDisplay.tsx:
```diff
@@ -0,0 +1,42 @@
+            name={geode + "_geode"}
+            size={30}
+            onClick={onGeodeClick}
+            title={Items.get_pretty_item_name(geode + "_geode")}
```
Instead of just showing the basic title html-property, I want to use the `useTooltip` hook to show more readable and useful tooltips. With different behaviours for regular, shift- and ctrl-clicking, it becomes even more important to show what will happen when you click. In `<GatheringBagDisplay />` you can take a look at how I have done it for the gathering bags which is probably very similar to how it can be done here.

sxe replied:
I agree, added it.

kape142 commented on src/overview/mining/MiningOverview.tsx:
```diff
@@ -1,24 +1,95 @@
+          size={30}
+          style={{
+            justifyContent: "center",
+            color: oilIn >= oilOut ? "#fff" : "#ff0000",
```
The color change here on negative flow was a great touch!

sxe replied:
Thanks. :)

kape142 commented on src/overview/mining/MiningOverview.tsx:
```diff
@@ -1,24 +1,95 @@
+  const [rocketFuel] = useNumberItemObserver("rocket_fuel", id);
+
+  const onRocketClick = (event: MouseEvent) => {
+    Modals.clicks_rocket();
```
Don't actually know the exact issue here, but when I click the rocket in the overview, nothing happens. When I click the rocket in the mining inventory i get a modal called 'OXYGEN REQUIRED', so maybe if you have the rocket but don't meet the other requirements for using the rocket (having an oxygen mask and an oxygen tank), the rocket can be visible but not clickable? Probably hard to test without having access to an account with the exact state described, so if you're not sure how to fix it I can take a look at it after merging your pr.

kape142 replied:
As mentioned I don't actually have access to send my rocket yet, so I can't test this, but if the modal is not doing much more than confirming that you want to send the rocket, I prefer if it can also be done without opening a modal, and then also with a tooltip using the useTooltip hook specifiying what will happen when you click.

sxe replied:
I tried to figure it out but failed. As you said, that specific case is hard to catch when you have already progressed past that point. If it still needs fixing for your case please add that yourself. 

The modal for the rocket is super important cause it's the only way to determine in what orbit the moon is and therefor how long the rocket will take to the moon. You have to check this on a regular basis so having this modal is essential.
 I wanted to have a little indicator that shows when the best time to launch the rock is reached. Sadly there is no easy way to get that information manually. 

When you call `websocket.send('CLICKS_ROCKET=0');` the response immediately opens the modal.
`case "OPEN_ROCKET_DIALOGUE":
     Modals.open_send_rocket_dialogue(values_array[0], values_array[1]);
break;`

I did some testing and was able to get the values but I don't see a way to suppress the modal opening. Maybe when you don't reuse the existing websocket connection you can avoid that but that's something for an other time.

kape142 commented on src/overview/mining/MiningOverview.tsx:
```diff
@@ -36,6 +107,37 @@ const MiningOverview = () => {
+            }}
+          >
+            <IPimg
+              name={"moonstone"}
```
This makes a lot of sense to have just as a modal, but it still deserves a real tooltip from `useTooltip`, look at e.g. `<BrewingOverview />` for the simplest possible use of the hook to just give it a description of what will happen when you click.

sxe replied:
Added

kape142 replied:
The tooltip is not appearing for me, looks like maybe the parent div is missing a `position: relative`?

kape142 commented on src/overview/woodcutting/LogDisplay.tsx:
```diff
@@ -9,6 +10,9 @@ const id = "LogDisplay";
 const LogDisplay = ({ log }: Props) => {
   const [amount] = useNumberItemObserver(log, id);
 
+  const onLogClick = (event: MouseEvent) => {
```
Similar to geodes, in my opinion the benefit of not having to open a new modal for every type of log you want to add is more valuable than the option to specify an exact amount. Modal can be kept as one of the click options if you prefer that, but it should be possible to add heat without opening the modal, and it should have a proper tooltip from `useTooltip` specifiying what will happen when you click.

sxe replied:
I agree, a tootip is important here.
Using the modal here was a conscious decision tho. I think it's super important to have destructive decisions like "burning all available logs" guarded by a confirmation dialog. 
As suggested I added alternative modes to the shortcuts tho. I think this way we have the best of both worlds.

As a side note. Personally I would prefer to have it like this with the bags as well, since I already miss clicked once and opened bags I didn't want to open. :) Maybe we can change that an other time.

kape142 replied:
Yeah, I think the way you solved it for logs now is great, and doing it like that for bags as well is something I also feel makes sense. I think the ideal solution would be to make it possible for users to choose per type of item what they want each type of click to do in settings, but that requires even more work and a proper settings page and stuff, but maybe sometime in the future that could be something to look into.

kape142 commented on vite.config.ts:
```diff
@@ -6,7 +6,7 @@ import { resolve } from "path";
 const name = "Idle Pixel Fixed";
 const namespace = "com.kape142.idlepixelfixed";
-const version = "1.0.2";
+const version = "1.0.3";
```
1.0.3 is an appropriate version for this I think, and you have changed it all places it needs to be changed 👍

sxe replied:
Perfect

kape142 reviewed (changes requested):
Lots of great additions here! The changes I want to see before merging this are mostly related to being able to bypass modals to do things effectively, and having tooltips showing what will happen when you click, so you can know what will happen without a confirmation modal.

sxe committed:
add tooltip and add alternative ctrl and shift click options to geodes in overview

sxe committed:
change rocket status from "none" to "Idle" when idling at home

sxe committed:
add minerals to mining tile, change tooltips to show above elements, add tooltips to logs, geodes and minerals

sxe committed:
add tooltip foot :)

sxe committed:
Add rocket and moonstone tooltip, refactor rocket

sxe commented:
I made all the suggested changes. 

Besides that I added "minerals" the the mining tabs cause geodes without minerals doesn't make much sense cause then you still would have to switch to the mining tab. Now you can just open all geodes and then handle the minerals as well. Both "converting to xp" and "crafting rings" options are available.

*edit: Oh almost forgot. Already told you in discord but just for completeness. I changed the tooltips so they are shown above the elements, to avoid them reaching out of the right side of the screen. 

Hope I fixed all your suggestions. 
Let me know what you think.

Cheers

kape142 commented on src/overview/farming/FarmingOverview.tsx:
```diff
@@ -86,7 +86,6 @@ const FarmingOverview = () => {
             display: "flex",
             justifyContent: "flex-start",
             flexWrap: "wrap",
-            overflowY: "auto",
```
If you have 17 or more seeds available (so more than 2 rows of 8) it no longer adds a scroll bar so you can see them all. With it set to "auto" it shouldn't add the scroll bar when it isn't needed, and while it doesn't look great with the scroll bar there, it looks worse with the 3rd row being partially hidden behind the farm plots.

kape142 commented on src/overview/mining/MiningOverview.tsx:
```diff
@@ -36,6 +75,41 @@ const MiningOverview = () => {
+              name={"moonstone"}
+              size={30}
+              onClick={onMoonstoneClick}
+              title={Items.get_pretty_item_name("Moonstone")}
```
We don't need the title property if we are using tooltips from `useTooltip`

kape142 commented on src/overview/mining/Tooltip.tsx:
If this is meant as a generic Tooltip that can be used many places, maybe it can be moved out of the mining folder? There is a `<Tooltip />` component in the util/tooltip folder which is just leftover code and not used anywhere, maybe this actually usable version could be moved to that file instead, and it can be used for more tooltips in the future?

kape142 commented on src/overview/mining/GeodeDisplay.tsx:
```diff
@@ -0,0 +1,70 @@
+    [
+      <Tooltip
+        text={`Crack ${amount} ` + Items.get_pretty_item_name(geode) + ` Geode(s)`}
+        {...tooltipProps}
```
I believe that sending in the tooltipProps here does nothing? The basic component `<Tooltip/>` does not accept a prop `geode`, so I don't think you need the tooltipProps const at all maybe?

kape142 commented on src/overview/woodcutting/WoodcuttingOverview.tsx:
```diff
@@ -30,17 +32,40 @@ const WoodcuttingOverview = () => {
+          <IPimg
+            name={"heat"}
+            size={20}
+            title={Items.get_pretty_item_name("heat")}
```
like with moonstone, this will always resolve to "Heat", so you can just write `title="Heat"`. Or even better if you move the basic `<Tooltip />` out from mining you could use that here with just the text "Heat" to make it look better.

kape142 reviewed (changes requested):
I really like the new look of the tooltips! Just some small fixes needed, and it should be ready to merge

kape142 commented on src/overview/woodcutting/WoodcuttingOverview.tsx:
```diff
@@ -30,17 +32,40 @@ const WoodcuttingOverview = () => {
-          <LogDisplay log={log} key={log} />
+          <LogDisplay
+            log={log} 
+            logHeat={Cooking.LOG_HEAT_MAP[log]}
```
This does not take into account achievements. You need to use `Cooking.getHeatPerLog(log)`

sxe committed:
fixed not appearing moonstone tooltip

sxe committed:
removed unnecessary tooltipProp

sxe committed:
change heat retrieval method

sxe committed:
refactor generic tooltip

sxe committed:
the farming and brewing tile can now grow in height if necessary

sxe commented:
I made all the mentioned changes.
The farming and and brewing (for the fav mode) tile can now grow in size to be able to see all elements when there are too many. 

Cheers
